GDJS/tests/tests/hot-reloader.js

```diff
@@ -36,6 +36,412 @@ describe('gdjs.HotReloader.deepEqual', () => {
+    width: 0,
+    height: 0,
+    numberProperties: [],
+    stringProperties: [],
+    initialVariables: [],
+  };
+
+  /** @type {InstanceData} */
+  const defaultChildInstance = {
+    persistentUuid: '',
+    layer: '',
+    locked: false,
+    name: 'MyChildObject',
+    x: 0,
+    y: 0,
+    angle: 0,
+    zOrder: 0,
+    customSize: false,
+    width: 0,
+    height: 0,
+    numberProperties: [],
+    stringProperties: [],
+    initialVariables: [],
+  };
+
+  /** @type {ObjectData & gdjs.CustomObjectConfiguration} */
+  const defaultCustomObject = {
+    type: 'MyExtension::MyCustomObject',
+    name: 'MyCustomObject',
+    behaviors: [],
+    variables: [],
+    effects: [],
+    content: {},
+    childrenContent: {},
+  };
+
+  /** @type {LayerData} */
+  const baseLayer = {
+    name: '',
+    visibility: true,
+    cameras: [],
+    effects: [],
+    ambientLightColorR: 127,
+    ambientLightColorB: 127,
+    ambientLightColorG: 127,
+    isLightingLayer: false,
+    followBaseLayerCamera: false,
+  };
+
+  /**
+   * Create and return a minimum working sprite object data.
+   * @internal
+   * @param {{name: string, image: string}} data
+   * @returns {gdjs.SpriteObjectData}
+   */
+  const createSpriteData = ({ name, image }) => {
+    return {
+      type: 'Sprite',
+      name,
+      behaviors: [],
+      variables: [],
+      effects: [],
+      animations: [
+        {
+          name: 'animation',
+          directions: [
+            {
+              sprites: [
+                {
+                  originPoint: { name: 'Origin', x: 0, y: 0 },
+                  centerPoint: {
+                    name: 'Center',
+                    x: 0,
+                    y: 0,
+                    automatic: true,
+                  },
+                  points: [],
+                  hasCustomCollisionMask: false,
+                  customCollisionMask: [],
+                  image,
+                },
+              ],
+              timeBetweenFrames: 0,
+              looping: false,
+            },
+          ],
+          useMultipleDirections: false,
+        },
+      ],
+      updateIfNotVisible: false,
+    };
+  };
+
+  /**
+   * Create and return a minimum working scene data.
+   * @internal
+   * @param {{instances?: Partial<InstanceData>[], objects?: ObjectData[]}} data
+   * @returns {LayoutData}
+   */
+  const createSceneData = ({ instances, objects }) => {
+    return {
+      layers: [baseLayer],
+      variables: [],
+      r: 0,
+      v: 0,
+      b: 0,
+      mangledName: 'Scene1',
+      name: 'Scene1',
+      stopSoundsOnStartup: false,
+      title: '',
+      behaviorsSharedData: [],
+      objects: objects || [
+        createSpriteData({ name: 'MyObject', image: 'ResourceA' }),
+      ],
+      instances: instances
+        ? instances.map((instance) => ({ ...defaultInstance, ...instance }))
+        : [],
+      usedResources: [],
+    };
+  };
+
+  /**
+   * Create and return a minimum working scene data.
```

Comment:
Careful about your comments:
```suggestion
   * Create and return a minimum working events-based object ("custom object") data.
```